engine/engine_test.go

```diff
@@ -786,6 +786,78 @@ func TestQueriesAgainstOldEngine(t *testing.T) {
 	+ on() group_left()
 	sum(http_requests_total{ns="nginx"})`,
 		},
+		// Result is correct but this likely fails due to https://github.com/golang/go/issues/12025.
+		// TODO(saswatamcode): Test NaN cases separately.
+		// {
+		// 	name: "scalar func with NaN",
+		// 	load: `load 30s
+		//  	http_requests_total{pod="nginx-1"} 1+1x15
+		//  	http_requests_total{pod="nginx-2"} 1+2x18`,
+		// 	query: `scalar(http_requests_total)`,
+		// },
```

Comment:
Maybe we can add a new test case which does not use `testutil`. Possibly a dedicated test for error cases. We can create a new issue/PR for this.

Reply:
Created issue #88! Not implementing in this PR as it might be complicated, will follow up with a future one. Keeping TODO comment. 🙂